File: netflix-prize-relational-example/nodejs/search-by-title.mjs
import { searchMoviesByTitle } from './netflix-dal.mjs'

if (process.argv.length !== 3) {
  console.log('Usage: search-by-title <query>')
  process.exit(1)
}

const query = process.argv[2]

const result = await searchMoviesByTitle(query)
if (result.length === 0) {
  console.log(`No movies match “${query}.”`)
}

result.forEach(movie => {
  console.log(`${movie.id} “${movie.title}” (${movie.year})`)
})

// Explicitly exit sync since we are asynchronous at this point.
process.exit(0)

File: netflix-prize-relational-example/python/search_by_title.py
import sys

from netflix_dal import search_movies_by_title

if len(sys.argv) != 2:
    print('Usage: search_by_title <query>')
    exit(1)

query = sys.argv[1]
result = search_movies_by_title(query)

if len(result) == 0:
    print(f'No movies match “{query}.”')
    exit(0)

for movie in result:
    print(f'{movie[0]} “{movie[2]}” ({movie[1]})')
